### PR TITLE
chore: rework what we consider a supported component with this change (we will check attributes values as well)

### PR DESCRIPTION
**Pull request by czp13**
## Description

Refactor, change what we consider supported component for auto fill.

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

**Comment by czp13** [comment]
All cases need to be tested.

**Comment by czp13** [comment]
I have tested and works, waiting for approvals now 🙇 .

File: src/test/java/com/vaadin/flow/ai/formfiller/views/FormFillerTextDemo.java [hunk 1]
```diff
@@ -216,8 +216,8 @@ public FormFillerTextDemo() {
         ComboBox<String> texts = new ComboBox<>("Select a text or just type your own <br>in the debug Input Source field");
         texts.setItems("Text1", "Text2", "Text3");
         texts.setValue("Text1");
-        texts.setAllowCustomValue(false);
         debugTool.getDebugInput().setValue(getExampleTexts().get("Text1"));
+        texts.setAllowCustomValue(false);
```
Comment (mshabarov) [inline src/test/java/com/vaadin/flow/ai/formfiller/views/FormFillerTextDemo.java:220]: Why is this needed?

**Review by mshabarov — CHANGES_REQUESTED** [review]
Needs a unit test for `getComponentInfo` that covers new conditions in `isSupportedAndAccepted`.

File: src/test/java/com/vaadin/flow/ai/formfiller/views/FormFillerTextDemo.java (hunk 1 above)
Comment (czp13) [reply src/test/java/com/vaadin/flow/ai/formfiller/views/FormFillerTextDemo.java:220]: Thanks for noticing it! 👍 

This is a mistake from me, I used it for testing the use cases, and then with my last commit, I cleaned up the testing code. 

But it looks like I added to the 220. line rather than the 219. line. I am fixing this!

File: src/main/java/com/vaadin/flow/ai/formfiller/utils/ComponentUtils.java [hunk 2]
```diff
@@ -239,6 +243,12 @@ public static void fillComponents(List<ComponentInfo> components, Map<String, Ob
                 logger.warn("Component has no id so it will be skipped: {}", componentInfo.component);
                 continue;
             }
+            if (!isSupportedComponent(componentInfo.component)) {
+                logger.error("Component type is not supported, this should have been discarded, " +
```
[the remark this answers is not in the document]
Comment (czp13) [reply src/main/java/com/vaadin/flow/ai/formfiller/utils/ComponentUtils.java:247]: Thank you!
Indeed, the `warn` level is much better, or even info as we can have many cases, which we do not want to auto-fill and those cases can be totally fine!

**Review by mshabarov — APPROVED** [review]
Looks good to me.
About unit tests, would you add a unit test here or add it to within another PR https://github.com/vaadin/form-filler-addon/pull/104 ?

**Comment by czp13** [comment]
> Looks good to me. About unit tests, would you add a unit test here or add it to within another PR #104 ?
I would add this to a separate PR if possible and sounds good to you! 🙇

**Review by mshabarov — APPROVED** [review]
